### PR TITLE
KOGITO-6998 Added configuring-openapi-services-endpoints guide

### PR DESCRIPTION
**Pull request by hbelmiro**
**JIRA:** https://issues.redhat.com/browse/KOGITO-6998

This PR depends on: https://github.com/kiegroup/kogito-examples/pull/1354

- [x] You have read the [contributions doc](https://github.com/kiegroup/kogito-docs/blob/main/CONTRIBUTING.md)
- [x] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [x] Pull Request title contains the target branch if not targeting main: `[0.9.x] KOGITO-XYZ Subject`
- [x] The nav.adoc file has a link to this guide in the proper category
- [x] The index.adoc file has a card to this guide in the proper category, with a meaningful description

### CODE REVIEW

**Review by fjtirado — COMMENTED** [review]
@ricardozanini 
I think this is missing configuration options provided by KOGITO-6887
Description of the example is nice, but somehow I feel the focus should be more on the properties thing.

**Comment by ricardozanini** [comment]
> @ricardozanini @hbelmiro I think this is missing configuration options provided by KOGITO-6687 Description of the example is nice, but somehow I feel the focus should be more on the properties thing.

I haven't read this guide yet, but yes [KOGITO-6687](https://issues.redhat.com/browse/KOGITO-6687) is crucial here.

The guide should help people to understand how they configure their endpoints, mainly in the cloud:

1. Understanding the property naming based on the OpenAPI file
2. Understanding that it can be overridden by ConfigMap and Env Properties
3. Understanding that the Kubernetes service endpoint can be used as the URL if the target service is within the same cluster. Example: `http://myservice.mynamespace.cluster.svc.local`

The question we are trying to answer here is, "How do I set the URLs that my workflow needs to call in runtime with little hurdle and annoying configuration properties?"

**Comment by hbelmiro** [comment]
> > @ricardozanini @hbelmiro I think this is missing configuration options provided by KOGITO-6687 Description of the example is nice, but somehow I feel the focus should be more on the properties thing.
> 
> I haven't read this guide yet, but yes [KOGITO-6687](https://issues.redhat.com/browse/KOGITO-6687) is crucial here.
> 
> The guide should help people to understand how they configure their endpoints, mainly in the cloud:
> 
> 1. Understanding the property naming based on the OpenAPI file
> 2. Understanding that it can be overridden by ConfigMap and Env Properties
> 3. Understanding that the Kubernetes service endpoint can be used as the URL if the target service is within the same cluster. Example: `http://myservice.mynamespace.cluster.svc.local`
> 
> The question we are trying to answer here is, "How do I set the URLs that my workflow needs to call in runtime with little hurdle and annoying configuration properties?"

@ricardozanini It seems like there's a misunderstanding here. The guide doesn't answer those questions.  I focused on using config profiles to configure OpenAPI in different environments. That was my understanding from:

> Instruct how to have different endpoint configuration profiles for OpenAPI calls
> Until we have a proper service discovery ([KOGITO-6697](https://issues.redhat.com/browse/KOGITO-6697)) implemented, we need a guide instructing users on how to configure their REST calls in different environments. One form of achieving that is configuring Quarkus profiles and mounting the application properties as CM in the cluster.

**Comment by ricardozanini** [comment]
These questions are part of the requirements to configure the target services.

**Comment by hbelmiro** [comment]
@fjtirado can you take a look at the "Configuring OpenAPI services endpoints" section where I explain config keys?
Commit: https://github.com/kiegroup/kogito-docs/pull/92/commits/77c302a509b2e299c5283e73fd6caaeeb4a4bd39

@ricardozanini @fjtirado this is still a work in progress.

**Comment by hbelmiro** [comment]
@ricardozanini @fjtirado 
PR ready to review.
Added information about https://issues.redhat.com/browse/KOGITO-6687, env variables, K8s ConfigMaps, and K8s service endpoint as URL.

**Review by ricardozanini — APPROVED** [review]

**Comment by hbelmiro** [comment]
@ricardozanini I renamed the guide since it has broader content now.

File: serverlessworkflow/modules/ROOT/pages/service-orchestration/configuring-openapi-services-endpoints.adoc [hunk 1]
```diff
@@ -1 +1,475 @@
...
+paths:
+  /stock-price/{symbol}:
+...
+----
+
+Kogito will use `"stock-service API"` as configuration key.
+
+[source, properties]
+----
+quarkus.rest-client."stock-service API".url=http://localhost:8282/
```
Comment (fjtirado) [inline serverlessworkflow/modules/ROOT/pages/service-orchestration/configuring-openapi-services-endpoints.adoc:120]: I think in this case (although Im not 100% sure) the property name will be
quarkus.rest-cluent.stock-service_API.url

File: serverlessworkflow/modules/ROOT/pages/service-orchestration/configuring-openapi-services-endpoints.adoc [hunk 2]
```diff
@@ -1 +1,475 @@
...
+    ]
+    ...
+}
+----
+
+Kogito will use `"myworkflow_myfunction"` as configuration key.
+
+[source, properties]
+----
+quarkus.rest-client."myworkflow_myfunction".url=http://localhost:8282/
```
Comment (fjtirado) [inline serverlessworkflow/modules/ROOT/pages/service-orchestration/configuring-openapi-services-endpoints.adoc:91]: quotes are not needed, property name is quarkus.rest-client.myworkflow_myfunction.url

File: serverlessworkflow/modules/ROOT/pages/service-orchestration/configuring-openapi-services-endpoints.adoc [hunk 3]
```diff
@@ -1 +1,475 @@
...
+* `FILE_NAME` - default value.
+Kogito will use a configuration key based on the OpenAPI document file name.
++
+--
+.Example
+Given an OpenAPI document located at `src/main/resources/openapi/stock-portfolio-svc.yaml`, Kogito will use `stock_portfolio_svc_yaml` as configuration key.
+
+[source,properties]
+----
+quarkus.rest-client."stock_portfolio_svc_yaml".url=http://localhost:8282/
```
Comment (fjtirado) [inline serverlessworkflow/modules/ROOT/pages/service-orchestration/configuring-openapi-services-endpoints.adoc:31]: quotes not needed, property name is
quarkus.rest-client.stock_portfolio_svc_yaml.url

File: serverlessworkflow/modules/ROOT/pages/service-orchestration/configuring-openapi-services-endpoints.adoc [hunk 4]
```diff
@@ -1 +1,475 @@
...
+    ...
+}
+----
+<1> URI path of the OpenAPI document.
+
+Kogito will use `"apicatalog_apis_123_document"` as configuration key.
+
+[source, properties]
+----
+quarkus.rest-client."apicatalog_apis_123_document".url=http://localhost:8282/
```
Comment (fjtirado) [inline serverlessworkflow/modules/ROOT/pages/service-orchestration/configuring-openapi-services-endpoints.adoc:60]: quoutes not needed, property name is 
quarkus.rest-client.apicatalog_apis_123_document.url

File: serverlessworkflow/modules/ROOT/pages/service-orchestration/configuring-openapi-services-endpoints.adoc (hunk 1 above)
Comment (fjtirado) [reply serverlessworkflow/modules/ROOT/pages/service-orchestration/configuring-openapi-services-endpoints.adoc:120]: Note that you can test this here

https://github.com/kiegroup/kogito-examples/tree/stable/serverless-workflow-examples/serverless-workflow-temperature-conversion

File: serverlessworkflow/modules/ROOT/pages/service-orchestration/configuring-openapi-services-endpoints.adoc (hunk 2 above)
Comment (hbelmiro) [reply serverlessworkflow/modules/ROOT/pages/service-orchestration/configuring-openapi-services-endpoints.adoc]: Done

File: serverlessworkflow/modules/ROOT/pages/service-orchestration/configuring-openapi-services-endpoints.adoc (hunk 4 above)
Comment (hbelmiro) [reply serverlessworkflow/modules/ROOT/pages/service-orchestration/configuring-openapi-services-endpoints.adoc]: Done

**Comment by hmanwani-rh** [comment]
Merging this PR to create patch PR with doc improvements.

**Pull request merged**